packages/core/src/Cardano/types/Cip1854PublicKey.ts

```diff
@@ -0,0 +1,55 @@
+import * as BaseEncoding from '@scure/base';
+import { InvalidStringError, OpaqueString, assertIsBech32WithPrefix, assertIsHexString } from '@cardano-sdk/util';
+
+const MAX_BECH32_LENGTH_LIMIT = 1023;
+export type Cip1854PublicKey = OpaqueString<'Cip1854PublicKey'>;
+
+export const Cip1854PublicKey = (value: string): Cip1854PublicKey => {
```

Comment:
It's using prefixes similar to Cardano addresses (starting with `stake_` and `addr_`). Addresses are not keys, they are have a credential (e.g. key **hash**) and also network id.

Also, what is the intended use case for this util? If it's for sharing the keys, then see

```suggestion
export const Cip1854Address = (value: string): Cip1854PublicKey => {
```

Reply:
This is a bit confusing on the CIP, but the CIP actually specifies these as keys not key hashes (plus network ids) (notice the _vk at the end).

Reply:
Thanks, resolving this :)